# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/BytecodeSection.cpp
# Commit: 8af30c6d9bb9
# Full Hash: 8af30c6d9bb91a1ed9a5bee712252bbe1bd20d85
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2020-11-14 09:46:25
# Regressor Bug: 1675241
# File Overlap Count: 1
# Description:
#   Bug 1675241 - Part 11: Use TaggedParserAtomIndex in ScriptThingVariant. r=tcampbell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96918
# ==============================================================================

diff -r cb0e58368cd4 -r 8af30c6d9bb9 js/src/frontend/BytecodeSection.cpp
--- a/js/src/frontend/BytecodeSection.cpp	Sat Nov 14 00:03:28 2020 +0000
+++ b/js/src/frontend/BytecodeSection.cpp	Sat Nov 14 00:03:36 2020 +0000
@@ -68,8 +68,8 @@
     uint32_t i;
     mozilla::Span<JS::GCCellPtr>& output;
 
-    bool operator()(const ScriptAtom& data) {
-      JSAtom* atom = data->toExistingJSAtom(cx, atomCache);
+    bool operator()(const TaggedParserAtomIndex& data) {
+      JSAtom* atom = atomCache.getExistingAtomAt(cx, data);
       MOZ_ASSERT(atom);
       output[i] = JS::GCCellPtr(atom);
       return true;